docs/build-macos.md

```diff
@@ -0,0 +1,52 @@
+macOS builds can be created using:
+
+- Clang, using Meson buildsystem and Homebrew for installing dependencies
+  (*recommended*)
+- GCC, using Meson buildsystem and Homebrew for installing dependencies.
+- Clang or GCC, using Meson buildsystem and MacPorts for dependencies.
+
+Only builds using Apple's Clang can link with Apple's Core SDK.
```

Comment:
MacOS builds can be created using the Meson build system with your choice of:
 - Clang or GCC for compiling, and
 - Brew or MacPorts for provision of dependencies.

We recommend Brew and Clang, and specifically the latter because only Apple's Clang can link with Apple's Core SDK libraries.

Reply:
(or "buildsystem", if that's what the Meson docs refer to it as)

Reply:
Another try, slightly more natural to read:
---
MacOS builds can be created using the Meson build system 
 - compiled using Clang or GCC having
 - dependencies provided by the Brew or MacPorts.

We recommend using Brew and Clang; specifically the latter because only Apple's Clang can link with Apple's Core SDK libraries.